sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -250,7 +250,7 @@ case class BroadcastQueryStageExec(
 
   override def cancel(): Unit = {
     if (!broadcast.relationFuture.isDone) {
-      sparkContext.cancelJobGroup(broadcast.runId.toString)
+      sparkContext.cancelJobsWithTag(broadcast.jobTag)
```

Comment:
hm,
currently BroadcastExchangeExec.doBroacast only does cancelJobsWithTag when it catches a TimeoutException.
Should we cancel there for broader exceptions (even any Throwable), and then `broadcast.relationFuture.cancel(true)` here should throw there and cancel there, and this place does not have to cancel separately?

Reply:
When AQE materialize broadcast relation, it does not call `BroadcastExchangeExec.doBroacast`, so it has no affect to cache exception in `BroadcastExchangeExec.doBroacast`.